templates/cache-backend.yaml

```diff
@@ -0,0 +1,148 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: 'Redis Cache for the Apollo server'
+
+Parameters:
+  VpcId:
+    Type: 'AWS::EC2::VPC::Id'
+
+  Subnets:
+    Type: 'List<AWS::EC2::Subnet::Id>'
+
+  Env:
+    Type: 'String'
+    Default: 'dev'
+
+  LogRetentionDays:
+    Type: 'Number'
+    Default: 7
+
+  CachePort:
+    Type: 'Number'
+    Default: 6379
+
+  NbrCacheClusters:
+    Type: 'Number'
+    Default: 1
+
+  AutomaticFailoverEnabled:
+    Type: 'String'
+    Default: 'false'
+    AllowedValues:
+      - 'true'
+      - 'false'
+
+  CacheNodeType:
+    Type: 'String'
+    Default: 'cache.t2.micro'
+
+  ClusterMode:
+    Type: 'String'
+    Default: 'Disabled'
+    AllowedValues:
+      - 'Disabled'
+      - 'Enabled'
+
+  Engine:
+    Type: 'String'
+    Default: 'valkey'
+    AllowedValues:
+      - 'redis'
+      - 'valkey'
+
+  EngineVersion:
+    Type: 'String'
+    Default: '7.2'
+
+Conditions:
+  FailoverEnabled:
+    !Equals [!Ref AutomaticFailoverEnabled, 'true']
+
+Resources:
+  CacheLogGroup:
+    Type: 'AWS::Logs::LogGroup'
+    Properties:
+      RetentionInDays: !Ref LogRetentionDays
+
+  # CacheSecurityGroup:
+  #   Type: 'AWS::ElastiCache::SecurityGroup'
+  #   Properties:
+  #     Description: !Sub "${AWS::StackName} SecGrp"
+
+  CacheSecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      GroupDescription: !Sub "${AWS::StackName} SecGrp"
+      GroupName: !Sub '${AWS::StackName}-cache-sec'
+      VpcId: !Ref VpcId
+
+  # CacheSecurityGroupIngress:
```

Comment:
L66 & L78 Is the commented out code left here for a reason?

Reply:
nope we ended up not needing it. I will remove